bittensor/metagraph.py

```diff
@@ -214,14 +224,79 @@ def uids(self) -> torch.LongTensor:
         return self.state.uids
 
     @property
-    def stake(self) -> torch.LongTensor:
+    def stake(self) -> torch.FloatTensor:
         r""" Returns the stake held by each known neuron.
         Returns
-            stake: (:obj:`torch.LongTensor` of shape :obj:`(n)`):
+            stake: (:obj:`torch.FloatTensor` of shape :obj:`(n)`):
+                stake of each known neuron.
+        """
+        return self.state.stake
+
+    @property
+    def S(self) -> torch.FloatTensor:
+        r""" Returns the stake held by each known neuron.
+        Returns
+            S: (:obj:`torch.FloatTensor` of shape :obj:`(n)`):
                 stake of each known neuron.
         """
         return self.state.stake
 
+    @property
+    def tau(self) -> torch.FloatTensor:
+        r""" tau: the chain per block inflation rate. i.e. 50
+        Returns
+            tau: (:obj:`torchFloatTensor` of shape :obj:`(1)`):
+                current chain inflation rate.
+        """
+        return self.state.tau
+
+    @property
+    def incentive(self) -> torch.FloatTensor:
+        r""" Returns the ranks 
+        Returns
+            incentive: (:obj:`torch.FLoatTensor` of shape :obj:`(n)`):
+                inflation incentive of each each known neuron.
+        """
+        I =  (self.tau * self.ranks) / torch.sum(self.ranks)
+        I = torch.where(torch.isnan(I), torch.zeros_like(I), I)
+        return I
+
+    @property
+    def I(self) -> torch.FloatTensor:
+        r""" Returns the inflation incentive for each peer per block.
+        Returns
+            I: (:obj:`torch.FloatTensor` of shape :obj:`(n)`):
+                stake of each known neuron.
+        """
+        return self.incentive
+
+    @property
+    def ranks(self) -> torch.FloatTensor:
+        r""" Returns the ranks W^t * S
+        Returns
+            ranks: (:obj:`torch.FloatTensor` of shape :obj:`(n)`):
+                rank of each known neuron.
+        """
+        if self.W.shape[0] == 0:
+            return torch.tensor([])
+        else:
+            # S.shape = [self.state.n]
+            # W.shape = [self.state.n, self.state.n]
+            # R.shape = [self.state.n]
+            S = self.S.view(self.state.n, 1)
+            W = torch.transpose(self.W.view(self.state.n, self.state.n), 0, 1)
+            R = torch.matmul(W, S).view(self.state.n)
+        return R
```

Comment:
The usage of S, W, and R this way is a bit confusing, can we rename the variables to something more descriptive?